src/services/inventoryService.ts

```diff
@@ -433,15 +433,19 @@ export const addMechSuit = async (mechsuitName: string, accountId: string) => {
 
 export const addSpecialItem = async (itemName: string, accountId: string) => {
     const inventory = await getInventory(accountId);
-    const specialItemIndex = inventory.SpecialItems.push({
-        ItemType: itemName,
-        Configs: [],
-        Features: 1,
-        UpgradeVer: 101,
-        XP: 0
-    });
-    const changedInventory = await inventory.save();
-    return changedInventory.SpecialItems[specialItemIndex - 1].toJSON();
+    // According to wiki there is one exalted item per suit type, so we check if we don't already have that item
```

Comment:
This is wrong, e.g. Titania has 2 exalted weapons (arch-gun & arch-melee).

Reply:
looks better now?